includes/GlobalFunctions.php

```diff
@@ -3808,28 +3808,32 @@ function wfWaitForSlaves( $wiki = false ) {
 	// bug 27975 - Don't try to wait for slaves if there are none
 	// Prevents permission error when getting master position
 	if ( $lb->getServerCount() > 1 ) {
-		/* Wikia change - added array() and $wiki parameters to getConnection to be able to wait for various DBs */
-		$dbw = $lb->getConnection( DB_MASTER, array(), $wiki );
-		$pos = $dbw->getMasterPos();
-		$lb->waitForAll( $pos, $wiki );
-
 		// Wikia change - begin
-		// OPS-6313 - sleep-based implementaion for consul-powered DB clusters
-		$masterHostName = $lb->getServerName(0);
-		if ( strpos( $masterHostName, '.service.consul' ) !== false ) {
-			# I am terribly sorry, but we do really need to wait for all slaves
-			# not just the one returned by consul
-			sleep( 1 );
+		// PLATFORM-1489: check if we're using consul configuration for DB slave
+		if ( $lb->hasConsulConfig() ) {
+			// get the list of IP addresses of all slave nodes from consul
+			// so that we can check all of them explicitly
+			$consul = new Wikia\Consul\Client();
```

Comment:
Just because we haven't tested something in the past doesn't mean we can't or shouldn't in the future. Using the DI layer allows us to begin to think about how we can test some of the core MW stuff. Is there any reason *not* to use it?